File: 2144-minimum-cost-of-buying-candies-with-discount/solution.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    int minimumCost(std::vector<int>& cost) {
        // Sort the costs in descending order
        std::sort(cost.rbegin(), cost.rend());
        
        int totalCost = 0;
        int n = cost.size();
        
        // Accumulate the cost of purchased candies, skipping every 3rd candy
        for (int i = 0; i < n; ++i) {
            if (i % 3 != 2) {
                totalCost += cost[i];
            }
        }
        
        return totalCost;
    }
};

File: 2144-minimum-cost-of-buying-candies-with-discount/solution.py
class Solution:
    def minimumCost(self, cost: list[int]) -> int:
        # Sort the costs in descending order
        cost.sort(reverse=True)
        
        total_cost = 0
        # Accumulate the cost of purchased candies, skipping every 3rd candy
        for i in range(len(cost)):
            if i % 3 != 2:
                total_cost += cost[i]
                
        return total_cost